misc/userscripts/ripbang

```diff
@@ -20,15 +20,17 @@ except ImportError:
 for argument in sys.argv[1:]:
     bang = '!' + argument
     r = requests.get('https://duckduckgo.com/',
-                     params={'q': bang + ' SEARCHTEXT'})
+                     params={'q': bang + ' SEARCHTEXT'},
+                     headers={'user-agent': 'ripbang'})
 
     searchengine = unquote(re.search("url=[^']+", r.text).group(0))
     searchengine = searchengine.replace('url=', '')
-    searchengine = searchengine.replace('/l/?kh=-1&uddg=', '')
+    searchengine = searchengine.replace('/l/?uddg=', '')
     searchengine = searchengine.replace('SEARCHTEXT', '{}')
+    searchengine = searchengine[0:searchengine.find("&rut")]
```

Comment:
Something like

```suggestion
    searchengine = searchengine.split("&rut")[0]
```

(untested)

would be a bit simpler and less prone to off-by-one errors.

Ideally, we'd use [`urllib.parse.urlparse`](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse), [`urllib.parse.parse_qs`](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.parse_qs) and [`urllib.parse.urlunparse`](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlunparse) to deal with the URL instead of handling it as a string... If you want to take a look at that, that'd be fantastic! But if you'd rather not, I'd be happy to take this in after the more minor changes, given that it's clearly an improvement.

Reply:
Alright, I've cleaned up the url grabbing a bit.